### PR TITLE
 Update timestamp parameter from sendInternal to be optional - Closes #7154 

### LINKED ISSUE #7154: Update `timestamp` parameter from `sendInternal` to be optional
### Expected behavior

`sendInternal` function in interoperability store does not require `timestamp` in sidechains but is used in the mainchain. 

We make the `timestamp` parameter optional so we do not have to pass in `Date.now` arbitrarily in sidechain context.

### Actual behavior

It currently requires `timestamp`. 

### Steps to reproduce

Check function parameters

### Which version(s) does this affect? (Environment, OS, etc...)

6.0

### PR DESCRIPTION
**Pull request by Incede**
### What was the problem?

This PR resolves #7154 

### How was it solved?

Update param & remove unnecessary usage

### How was it tested?

Updated test

### CODE REVIEW

**Review by mitsuaki-u — APPROVED** [review]
LGTM!

File: framework/src/modules/interoperability/mainchain/store.ts [hunk 1]
```diff
@@ -161,7 +161,8 @@ export class MainchainInteroperabilityStore extends BaseInteroperabilityStore {
 		// Chain must be live; This check is always on the receivingChainID
 		const isReceivingChainLive = await this.isLive(
 			receivingChainIDAsStoreKey,
-			sendContext.timestamp,
+			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+			sendContext.timestamp!,
 		);
```
Comment (ishantiw) [inline framework/src/modules/interoperability/mainchain/store.ts:166]: maybe its better to check if timestamp is undefined and throw error in isLive function in mainchain as its needed to check for the liveness condition

**Review by ishantiw — CHANGES_REQUESTED** [review]
LGTM, one comment related to timestamp check in isLive function

File: framework/src/modules/interoperability/mainchain/store.ts (hunk 1 above)
Comment (ishantiw) [reply framework/src/modules/interoperability/mainchain/store.ts]: isLive is being called from other places too, so maybe its better to throw it from one place

Comment (Incede) [reply framework/src/modules/interoperability/mainchain/store.ts]: other places don't have optional param,  no ?

Comment (ishantiw) [reply framework/src/modules/interoperability/mainchain/store.ts]: ah true, sorry ignore my comment

**Review by ishantiw — APPROVED** [review]
Thanks for the quick fix 👏

**Pull request merged**